### ISSUE TITLE
`imgpkg copy` prints completion for each donwloaded layer twice

### ISSUE THREAD

**Issue by pivotaljohn** [carvel-dev/imgpkg#11]
Got this:

```shell
$ imgpkg copy -b k8slt/webservers-demo:0.0.1 --to-tar=the-football.tar
copy | exporting 2 images...
copy | will export index.docker.io/k8slt/webservers-demo@sha256:5ebbd85028685b3f343fb6736109441643f2f85f59c919774c02d29943185477
copy | will export index.docker.io/library/nginx@sha256:b4b9b3eee194703fc2fa8afa5b7510c77ae70cfba567af1376a573a967c03dbb
copy | exported 2 images
copy | writing layers...
copy | done: file 'manifest.json' (40.388µs)
copy | done: file 'sha256-01a0caaddf4b559a9603a60e9cdd072d8b405d7c78a0a83a92e834b3c0898733.tar.gz' (9.126µs)
copy | done: file 'sha256-0a4690c5d889e116874bf45dc757b515565a3bd9b0f6c04054d62280bb4f4ecf.tar.gz' (14.790296ms)
copy | done: file 'sha256-0e75755515daa100dcb232b335d5b61d0278b5850bf0992f2927447031769a3e.tar.gz' (15.836402ms)
copy | done: file 'sha256-2eead4197fac409644fd8aaf115559d6383b0d56f1ad04d7116aaabbcbea8bed.tar.gz' (36.950171ms)
copy | done: file 'sha256-44446b456159d28726ce1ce5eaf894fa98415dce50d113ffdcc6cb99505bd109.tar.gz' (5.527µs)
copy | done: file 'sha256-458117d4230ebcca9a649824e2a9a8d8a7c3b39ee0cca6ae57cc11e0debe713e.tar.gz' (3.863µs)
copy | done: file 'sha256-45d6b85619b1cdde5c938d82a688d73c5532d0d9992b1af33d4a47aa09e0b699.tar.gz' (20.615259ms)
copy | done: file 'sha256-494350645e1c39e92e069b34b7ce0310dc6496d163f078bb1400391a3a574105.tar.gz' (27.39677ms)
copy | done: file 'sha256-55541b6f4770186b6d3e005cd06a9508301ed3c92877d4e627faed87923ffdbc.tar.gz' (4.93µs)
copy | done: file 'sha256-5e1d4775cd524fabdf3b94fcf2a517eb62c74ff55ccf7c32779218df94026295.tar.gz' (3.844µs)
copy | done: file 'sha256-7f05216ff691c8948e167b003b68728fe14854394d4f1287224705f9487985e4.tar.gz' (26.332055ms)
copy | done: file 'sha256-9719afee3eb7680a805449b50040fed1fff0d065f58d186e6e2bb4959f3d9e6e.tar.gz' (34.64653ms)
copy | done: file 'sha256-989ec9c2d5c0faa3c30e41cf0fbdb98632d8d5ef12aa71801d619454a65a3df3.tar.gz' (21.506775ms)
copy | done: file 'sha256-9a806ca4c22e0f67425c88867deec63e29dbd1463d60e4bf4ec32ef8af0fa174.tar.gz' (5.423µs)
copy | done: file 'sha256-af822a1ba456a0c6c2c0e954103b6b630ca93b8ea69cd8c34496d0149c94b27b.tar.gz' (24.455231ms)
copy | done: file 'sha256-b6857afb67f0f43f41d622a660c1164719c1622b528284dd739a4b35e7bb07e4.tar.gz' (4.368µs)
copy | done: file 'sha256-d8f0c6f8a424139159764e11d49d9ad098fc72ebb8397f31b6d26e13545f9a04.tar.gz' (24.878672ms)
copy | done: file 'sha256-f01e2fc672b937f58be589455802ade04da5bae6a188a46a6090252055118d90.tar.gz' (36.67724ms)
copy | done: file 'sha256-fa05dcee3c0c86d4f7c7198c7083a2d314b6bfa3fb483e9c6d80317c5d7fe09d.tar.gz' (41.844033ms)
copy | done: file 'sha256-44446b456159d28726ce1ce5eaf894fa98415dce50d113ffdcc6cb99505bd109.tar.gz' (64.148µs)
copy | done: file 'sha256-d8f0c6f8a424139159764e11d49d9ad098fc72ebb8397f31b6d26e13545f9a04.tar.gz' (8.689763164s)
copy | done: file 'sha256-f01e2fc672b937f58be589455802ade04da5bae6a188a46a6090252055118d90.tar.gz' (7.20779871s)
copy | done: file 'sha256-458117d4230ebcca9a649824e2a9a8d8a7c3b39ee0cca6ae57cc11e0debe713e.tar.gz' (28.163µs)
copy | done: file 'sha256-494350645e1c39e92e069b34b7ce0310dc6496d163f078bb1400391a3a574105.tar.gz' (2.572949236s)
copy | done: file 'sha256-b6857afb67f0f43f41d622a660c1164719c1622b528284dd739a4b35e7bb07e4.tar.gz' (30.823µs)
copy | done: file 'sha256-01a0caaddf4b559a9603a60e9cdd072d8b405d7c78a0a83a92e834b3c0898733.tar.gz' (81.487µs)
copy | done: file 'sha256-af822a1ba456a0c6c2c0e954103b6b630ca93b8ea69cd8c34496d0149c94b27b.tar.gz' (15.296891022s)
copy | done: file 'sha256-fa05dcee3c0c86d4f7c7198c7083a2d314b6bfa3fb483e9c6d80317c5d7fe09d.tar.gz' (6.293347155s)
copy | done: file 'sha256-55541b6f4770186b6d3e005cd06a9508301ed3c92877d4e627faed87923ffdbc.tar.gz' (33.065µs)
copy | done: file 'sha256-0a4690c5d889e116874bf45dc757b515565a3bd9b0f6c04054d62280bb4f4ecf.tar.gz' (17.881268948s)
copy | done: file 'sha256-9a806ca4c22e0f67425c88867deec63e29dbd1463d60e4bf4ec32ef8af0fa174.tar.gz' (29.568µs)
copy | done: file 'sha256-7f05216ff691c8948e167b003b68728fe14854394d4f1287224705f9487985e4.tar.gz' (4.82642281s)
copy | done: file 'sha256-9719afee3eb7680a805449b50040fed1fff0d065f58d186e6e2bb4959f3d9e6e.tar.gz' (22.618497646s)
copy | done: file 'sha256-5e1d4775cd524fabdf3b94fcf2a517eb62c74ff55ccf7c32779218df94026295.tar.gz' (31.008µs)
copy | done: file 'sha256-989ec9c2d5c0faa3c30e41cf0fbdb98632d8d5ef12aa71801d619454a65a3df3.tar.gz' (5.397030992s)
copy | done: file 'sha256-0e75755515daa100dcb232b335d5b61d0278b5850bf0992f2927447031769a3e.tar.gz' (6.610824467s)
copy | done: file 'sha256-2eead4197fac409644fd8aaf115559d6383b0d56f1ad04d7116aaabbcbea8bed.tar.gz' (5.568175778s)
copy | done: file 'sha256-45d6b85619b1cdde5c938d82a688d73c5532d0d9992b1af33d4a47aa09e0b699.tar.gz' (5.627179799s)
```

when

```shell
$ tar tvf the-football.tar | awk '{print $9}' | sort
manifest.json
sha256-01a0caaddf4b559a9603a60e9cdd072d8b405d7c78a0a83a92e834b3c0898733.tar.gz
sha256-0a4690c5d889e116874bf45dc757b515565a3bd9b0f6c04054d62280bb4f4ecf.tar.gz
sha256-0e75755515daa100dcb232b335d5b61d0278b5850bf0992f2927447031769a3e.tar.gz
sha256-2eead4197fac409644fd8aaf115559d6383b0d56f1ad04d7116aaabbcbea8bed.tar.gz
sha256-44446b456159d28726ce1ce5eaf894fa98415dce50d113ffdcc6cb99505bd109.tar.gz
sha256-458117d4230ebcca9a649824e2a9a8d8a7c3b39ee0cca6ae57cc11e0debe713e.tar.gz
sha256-45d6b85619b1cdde5c938d82a688d73c5532d0d9992b1af33d4a47aa09e0b699.tar.gz
sha256-494350645e1c39e92e069b34b7ce0310dc6496d163f078bb1400391a3a574105.tar.gz
sha256-55541b6f4770186b6d3e005cd06a9508301ed3c92877d4e627faed87923ffdbc.tar.gz
sha256-5e1d4775cd524fabdf3b94fcf2a517eb62c74ff55ccf7c32779218df94026295.tar.gz
sha256-7f05216ff691c8948e167b003b68728fe14854394d4f1287224705f9487985e4.tar.gz
sha256-9719afee3eb7680a805449b50040fed1fff0d065f58d186e6e2bb4959f3d9e6e.tar.gz
sha256-989ec9c2d5c0faa3c30e41cf0fbdb98632d8d5ef12aa71801d619454a65a3df3.tar.gz
sha256-9a806ca4c22e0f67425c88867deec63e29dbd1463d60e4bf4ec32ef8af0fa174.tar.gz
sha256-af822a1ba456a0c6c2c0e954103b6b630ca93b8ea69cd8c34496d0149c94b27b.tar.gz
sha256-b6857afb67f0f43f41d622a660c1164719c1622b528284dd739a4b35e7bb07e4.tar.gz
sha256-d8f0c6f8a424139159764e11d49d9ad098fc72ebb8397f31b6d26e13545f9a04.tar.gz
sha256-f01e2fc672b937f58be589455802ade04da5bae6a188a46a6090252055118d90.tar.gz
sha256-fa05dcee3c0c86d4f7c7198c7083a2d314b6bfa3fb483e9c6d80317c5d7fe09d.tar.gz
```

Is there a way to either mate-up each "copy from" with its "copy to"? or clarify the source/sink involved in a given line of output?

**Comment by cppforlife** [comment]
im not so sure its about printing. are we writing each layer twice?

**Comment by pivotaljohn** [comment]
I'm inferring from the timings that the first mention is when we copy _from_ the registry and the second mention is when we copy _to_ the tarball.

**Comment by cppforlife** [comment]
> we copy from the registry and the second mention is when we copy to the tarball.

http stream should be going directly into tarball. there is no intermediate.

**Comment by cppforlife** [comment]
fixed in kbld by https://github.com/k14s/kbld/commit/dc40e8951749bedfed3891b39ad022976316e1f8.

**Comment by ewrenn8** [comment]
fixed in 998cb7e7850a1c69b2d2fae59ca9e4ca3edcbb03
